component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/voiceotp/VoiceOTPUtils.java

```diff
@@ -0,0 +1,562 @@
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.application.common.model.Property;
+import org.wso2.carbon.identity.authenticator.voiceotp.exception.VoiceOTPException;
+import org.wso2.carbon.identity.authenticator.voiceotp.internal.VoiceOTPServiceDataHolder;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.user.api.UserRealm;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.user.api.UserStoreManager;
+import org.wso2.carbon.user.core.service.RealmService;
+import org.wso2.carbon.utils.multitenancy.MultitenantUtils;
+
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * VoiceOTPUtils class.
+ */
+public class VoiceOTPUtils {
+
+    private static final Log log = LogFactory.getLog(VoiceOTPUtils.class);
+    private static boolean useInternalErrorCodes = true;
+
+    /**
+     * Get parameter values from application-authentication.xml local file.
+     */
+    public static Map<String, String> getVoiceParameters() {
+
+        AuthenticatorConfig authConfig = FileBasedConfigurationBuilder.getInstance()
+                .getAuthenticatorBean(VoiceOTPConstants.AUTHENTICATOR_NAME);
+        if (authConfig != null) {
+            return authConfig.getParameterMap();
+        }
+
+        log.debug("Authenticator configs not found. Hence returning an empty map");
+
+        return Collections.emptyMap();
+    }
+
+    /**
+     * Check whether VoiceOTP is disabled by user.
+     *
+     * @param username the Username.
+     * @param context  the AuthenticationContext.
+     * @return true or false.
+     * @throws VoiceOTPException Exception.
+     */
+    public static boolean isVoiceOTPDisableForLocalUser(String username, AuthenticationContext context)
+            throws VoiceOTPException {
+
+        UserRealm userRealm;
+        try {
+            String tenantDomain = MultitenantUtils.getTenantDomain(username);
+            int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+            RealmService realmService = IdentityTenantUtil.getRealmService();
+            userRealm = realmService.getTenantUserRealm(tenantId);
+            username = MultitenantUtils.getTenantAwareUsername(String.valueOf(username));
+            boolean isEnablingControlledByUser = isVoiceOTPEnabledByUser(context);
+            if (userRealm != null) {
+                if (isEnablingControlledByUser) {
+                    Map<String, String> claimValues = userRealm.getUserStoreManager().getUserClaimValues(username,
+                            new String[]{VoiceOTPConstants.USER_VOICEOTP_DISABLED_CLAIM_URI}, null);
+                    return Boolean.parseBoolean(claimValues.get(VoiceOTPConstants.USER_VOICEOTP_DISABLED_CLAIM_URI));
+                }
+            } else {
+                throw new VoiceOTPException("Cannot find the user realm for the given tenant domain : " + tenantDomain);
+            }
```

Comment:
Let's change the condition of the if statement to its negation for better code readability